tiledb/sm/query/query_condition.cc

```diff
@@ -1470,13 +1516,87 @@ Status QueryCondition::apply_clause_sparse(
     case Datatype::STRING_UCS2:
     case Datatype::STRING_UCS4:
     default:
-      return Status_QueryConditionError(
-          "Cannot perform query comparison; Unsupported query "
-          "conditional type on " +
-          clause.field_name_);
+      throw std::runtime_error(
+          "Cannot perform query comparison; Unsupported query conditional type "
+          "on " +
+          node->get_field_name());
   }
+}
 
-  return Status::Ok();
+template <typename BitmapType, typename CombinationOp>
+void QueryCondition::apply_tree_sparse(
+    const tdb_unique_ptr<ASTNode>& node,
+    const ArraySchema& array_schema,
+    ResultTile& result_tile,
```

Comment:
if this is an input (aka read-only) argument this should be a `const` reference.
If it's an output or input/output argument we should better move it to the end of the argument list where output arguments are found by convention so that it's easier for the reader to tell the purpose.

Reply:
This is because the function` ResultTile::tile_tuple` is not marked const. I am unsure if marking it const would change anything. Can you/someone else who knows the structure better than myself take a look at `tiledb/sm/query/result_tile.cc` check that marking the funciton as const is okay?

Reply:
Ok, this means that we want result_tile to be modifiable, so you can leave it as is - it's already in the end of the argument list together with `result_bitmap`.